tests/test_config.py

```diff
@@ -19,31 +19,36 @@
 plugins:
   generators:
     huggingface:
-      dtype: general
-      gpu: 0
+      hf_args:
+        torch_dtype: float16
       Pipeline:
-        dtype: bfloat16
+        hf_args:
+            device: cuda
   probes:
     test:
       generators:
         huggingface:
             Pipeline:
-              dtype: for_probe
+                hf_args:
+                    torch_dtype: float16
   detector:
       test:
         val: tests
         Blank:
           generators:
             huggingface:
-                gpu: 1
+                hf_args:
+                    torch_dtype: float16
+                    device: cuda:1
```

Comment:
how come `gpu: 1` changed to `cuda:1` here...

Reply:
These are simply example format being documented in the test file. They are not actually `consumed` values in the tests.

Reply:
Can we increase consistency to make the example more straightforward?

Reply:
I would suggest the goal of the example is to show what you might configure differently depending on context.

The example here would define a model under test that is a `huggingface.Pipeline` would used any `cuda` devices while a `huggingface` generator created by for use by the `test.Blank` detector would specify to use `cuda:1` for it's device, and a `huggingface` generator created by for use by the `test.Blank` buff would use device `cuda:0`.  This would mean the model under tests is free to use any available device while the detector and buff would use specific different GPU devices from each other.

I think a more `concrete` or `viable` example might be on the horizo. This level of detail in config is likely rare for users to need and @erickgalinkin may be the only actual consumer for this level for the near term. 😄

Reply:
This feels right -- I suspect the overwhelming majority of users will not need to configure things like this, but it's good to have. I'll likely need to refactor parts of `AutoDAN` and/or `TAP` as a result of this PR being merged and it's plausible that we'll need this level of granularity in configs, particularly for folks using resource-constrained systems.